homeworks/08-functional-reactive-programming/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import { fromEvent, interval } from 'rxjs';
+import {
+  map, scan, startWith, switchMap, takeWhile, tap,
+} from 'rxjs/operators';
+import {
+  dog, moveDot, setGameOverText, setTimerText, time, updatedScore,
+} from './render';
+import './style.css';
+import '../assets/cursor.svg';
+import '../assets/dog.jpg';
+import './index.html';
+import '../assets/grass.png';
+
+interface State {
+    score: number;
+    time: number;
+}
+
+const gameState: State = { score: -1, time: 500 };
+
+const makeInterval = (val: State) => interval(val.time).pipe(
+  map((v) => `${time - v}`),
```

Comment:
Don't cast the value to string here please. It allows you to not cast the value again to number inside `isNotGameOver`.